cmd/lotus-chainwatch/syncer/sync.go

```diff
@@ -257,9 +250,8 @@ func (s *Syncer) unsyncedBlocks(ctx context.Context, head *types.TipSet, since t
 	return toSync, nil
```

Comment:
I think there's a bug in this for-loop. If we sync to 2000. Then turn Sentinel off for another 2000, the syncedBlocks func will not return anything. We probably need to consider `since` as part of the chain-walking loop. 

Something like:
```go
for _, header := range pts.Blocks() {
  if header.Height > since {
    toVisit.PushBack(header)
  }
}
```

Reply:
May have spoken too soon. I misunderstood the SQL in syncedBlocks by thinking the join might allow capture of s.lookBack NULL rows, but the height and `IS NOT NULL` conditions should be sufficient. Nevermind this.